src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/.template.config/template.json

```diff
@@ -261,7 +349,7 @@
       "defaultValue": "InteractivePerPage",
       "displayName": "_Interactivity location",
       "description": "Chooses which components will have interactive rendering enabled",
-      "isEnabled": "(InteractivityPlatform != \"None\")",
+      "isEnabled": "(InteractivityPlatform != \"None\" && auth == \"None\")",
```

Comment:
Just to be clear, this doesn't work in the way it should in VS. It will still present the option to pick a value for this, but will then ignore it if you have a nonempty auth value. It will be pretty confusing.

Not saying there's anything else you can do right now. Just making sure it's not a surprise later.

Reply:
Right. It's the exact same thing that already happens if you set the "Interactivity type" to "None" and try to set the "Interactivity location" to "Global". The "Interactivity location" gets silently ignored and you get the "Per page/component" behavior.

I did briefly try to make server-only global interactivity work, considering that's the only form of global interactivity that had a hope of working since it's actually possible to route to components running on the server which is required by the Identity Components (not to mention having the `MainLayout` defined). Unfortunately, not even that worked because any component with an `HttpContext` `[CascadingParameter]` gets null in that case. I intend to file a bug report, but I have been busy with these template changes.

![image](https://github.com/dotnet/aspnetcore/assets/54385/e911de7c-c677-4770-bec8-1e7ee369108f)

Reply:
> Unfortunately, not even that worked because any component with an HttpContext [CascadingParameter] gets null in that case. I intend to file a bug report, but I have been busy with these template changes.

That's not a bug, since during interactive server rendering, you're on a SignalR connection and there isn't necessarily any `HttpContext`. There certainly isn't an `HttpResponse` you can do things with.

To work in interactive server rendering, we would want either:

 * Some way to exclude these endpoints from discovery by the `Router`, so navigating to them would naturally become a full page load to a noninteractive SSR page.
 * Or, better, make them work from within the circuit by triggering all page loads that set cookies etc on an individual bases for each interaction that requires it